m4/ax_with_libdeflate.m4

```diff
@@ -0,0 +1,96 @@
+# SYNOPSIS
+#
+#   AX_WITH_LIBDEFLATE([ACTION-IF-TRUE], [ACTION-IF-FALSE])
+#
+# DESCRIPTION
+#
+#   This macro checks whether Libdeflate is installed and adds a
+# --with-libdeflate=DIR option to override the search path.
+# See https://github.com/ebiggers/libdeflate for the library itself.
+#
+#   The following output variables are amended by this macro:
+#
+#     CPPFLAGS     Preprocessor flags for compiling
+#     LDFLAGS      Linker flags for linking against the library
+#     LIBS         Library list
+#
+#   It also sets LIBDEFLATE_LDFLAGS variable, to aid creation of
+#   pkg-config files.
+#
+#   The HAVE_LIBDEFLATE cpp variable will be defined in a working
+#   libdeflate was found.
+#
+# LICENSE
+#
+#   Copyright (C) 2018 Genome Research Ltd
+#
+#   Copying and distribution of this file, with or without modification, are
+#   permitted in any medium without royalty provided the copyright notice
+#   and this notice are preserved.  This file is offered as-is, without any
+#   warranty.
+AC_DEFUN([AX_LIBDEFLATE],
+[
+  AC_ARG_WITH(libdeflate,
+	      AC_HELP_STRING([--with-libdeflate=DIR],[look for libdeflate in DIR]),
+	      [_libdeflate_with=$withval],[_libdeflate_with="no"])
+
+  # Check if it's a working library
+  libdeflate_ok=no
+  _cppflags=$CPPFLAGS
+  _ldflags=$LDFLAGS
+  if test "x$_libdeflate_with" != "xno"
+  then
+    if test "$_libdeflate_with" != "yes"
+    then
+      if test -f "${_libdeflate_with}/include/libdeflate.h"
```

Comment:
The original PR didn't have any configure script at all because I knew you'd object to the way I do it, but after languishing for 6 months I decided to move it along and do it.  GNU themselves aren't sure of how to do this, hence I did it my way.

1) This is compatible with the wording of the autoconf manual, which states "An argument that is neither ‘yes’ nor ‘no’ could include a name or number of a version of the other package, to specify more precisely which other package this program is supposed to work with".  A directory is a very precise way of indicating the package to work with and at no point does it forbid this.  https://www.gnu.org/software/autoconf/manual/autoconf-2.66/html_node/External-Software.html#External-Software

2) GNU themselves also recommend use of it in this manner.  Eg https://www.gnu.org/software/libidn/manual/html_node/Autoconf-tests.html

3) It's *useful*.  Let's face it, people aren't going to find libdeflate on the system automatically.  I could understand the argument for something common and available for install on most systems, but custom software is basically going to boil down to manually specifying LIBS, CPPFLAGS and LDFLAGS.  In which case you may as well also add CFLAGS=-DHAVE__LIBDEFLATE and ignore the whole autoconf in the first place for what little gain it gives you.

4) If you really dislike this, then I assume you'll be modifying samtools configure.ac to disallow --with-htslib=DIR too?  There is precedent within this package.

5) Specifying a directory isn't mandatory.  You could always use `--with-libdeflate=yes` and then add in the directory it resides, twice (once for CPPFLAGS and once for LDFLAGS), if you enjoy typing. :-)